stdlib/public/Reflection/TypeRefBuilder.cpp

```diff
@@ -193,36 +193,85 @@ const TypeRef *TypeRefBuilder::lookupSuperclass(const TypeRef *TR) {
   return Unsubstituted->subst(*this, *SubstMap);
 }
 
+static llvm::Optional<StringRef> FindOutermostModuleName(NodePointer Node) {
+  // Breadth first search until we find the module name so we find the outermost
+  // one.
```

Comment:
do we know if this guaranteed, that the module a type is defined in will always be the shallowest Module node?

Reply:
That's how I've always understood it, maybe @jckarter can confirm this for us?

Reply:
Module nodes should only ever be root nodes.